.github/workflows/build-portable-mlir-toolchain.yml

```
@@ -0,0 +1,58 @@
+name: Build Portable MLIR Toolchain (staged)
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: "llvm-project ref or tag (e.g., llvmorg-21.1.0)"
+        required: true
+        type: string
+      release:
+        description: "Whether to create a GitHub Release with the built artifacts"
+        required: true
+        type: boolean
+      tag:
+        description: "Tag for the GitHub Release"
+        required: false
+        type: string
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  linux:
+    uses: ./.github/workflows/build-mlir-linux.yml
+    with:
+      ref: ${{ inputs.ref || 'llvmorg-21.1.5' }}
+
+  macos:
+    uses: ./.github/workflows/build-mlir-macos.yml
+    with:
+      ref: ${{ inputs.ref || 'llvmorg-21.1.5' }}
+
+  windows:
+    uses: ./.github/workflows/build-mlir-windows.yml
+    with:
+      ref: ${{ inputs.ref || 'llvmorg-21.1.5' }}
+
+  release:
+    needs: [linux, macos, windows]
+    if: ${{ inputs.release == true || true }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5.0.0
+      - name: Download artifacts
+        uses: actions/download-artifact@v5.0.0
+        with:
+          path: artifacts
+      - name: Create GitHub Release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create ${{ inputs.tag || 'test-release-6'}} \
```

Just as a reminder, before it makes the next run fail. 😌 

```suggestion
          gh release create ${{ inputs.tag || 'test-release-7'}} \
```